e2e_tests/conftest.py

```diff
@@ -36,19 +131,42 @@ def _create_user_page(username, password) -> Page:
     return _create_user_page
 
 
+@pytest.fixture(scope="session")
+def create_users():
+    client_aau = KeycloakClient("all_access_user")
```

Comment:
Do you think the following is simpler?

```
@pytest.fixture
def keycloak_admin():
        realm_name = os.environ.get("KEYCLOAK_REALM_NAME")
        client_id = os.environ.get("KEYCLOAK_CLIENT_ID")
        client_secret = os.environ.get("KEYCLOAK_CLIENT_SECRET")
        server_url = os.environ.get("KEYCLOAK_BASE_URI")

        keycloak_openid = keycloak.KeycloakOpenID(
            server_url=server_url,
            client_id=client_id,
            realm_name=realm_name,
            client_secret_key=client_secret,
        )

        token = keycloak_openid.token(grant_type="client_credentials")
        keycload_admin = keycloak.KeycloakAdmin(
            server_url=server_url,
            realm_name=realm_name,
            token=token,
        )
        return keycload_admin


@pytest.fixture
def create_keycloak_user(keycloak_admin):
    def _create_keycloak_user(groups):
            user_email = f"{uuid.uuid4().hex}{user_type}@test.com"
            user_pass = uuid.uuid4().hex
            user_first_name = "Test"
            user_last_name = "Name"
            user_id = keycloak_admin.create_user(
                {
                    "firstName": user_first_name,
                    "lastName": user_last_name,
                    "username": user_email,
                    "email": user_email,
                    "enabled": True,
                    "groups": groups,
                    "credentials": [{"value": user_pass, "type": "password"}],
                }
            )
            return user_id, user_email, user_pass
    return _create_keycloak_user

@pytest.fixture
def create_temporary_users_for_e2e_tests(keycloak_admin):
    all_access_user_groups = ["/ayr_user_type/view_all"]
    testing_a_user_groups = ["/ayr_user_type/view_dept","/transferring_body_user/Testing A"]

    aau_user_id, aau_user_email, aau_user_pass = create_keycloak_user(all_access_user_groups)
    testing_a_user_id, testing_a_user_email, testing_a_user_pass = create_keycloak_user(testing_a_user_groups)

    yield {
        "aau": {
            "username": aau_user_email,
            "password": aau_user_pass,
        },
        "standard": {
            "username": testing_a_user_email,
            "password": testing_a_user_pass,
        },
    }

    keycloak_admin.delete_user(user_id=aau_user_id)
    keycloak_admin.delete_user(user_id=testing_a_user_id)
```


But at minimum I think the client should not need to know about the type of user, instead "all_access_user" or just equivalent groups could be passed to `create_user`.

Also there are soe variables unused which I have removed in the rewrite above but if we keep it in the current format we should remove, for example `auth_url`